torch/csrc/jit/codegen/cuda/ir_iostream.cpp

```diff
@@ -54,13 +45,48 @@ void IrPrinter::handle(const Statement* s) {
 }
 
 void IrPrinter::handle(const Val* v) {
-  OptInConstDispatch::handle(v);
+  if (print_inline_) {
+    os_ << v->toInlineString(indent_size_);
+  } else {
+    os_ << v->toString(indent_size_);
+  }
 }
 
 void IrPrinter::handle(const Expr* e) {
   OptInConstDispatch::handle(e);
 }
 
+void IrPrinter::handle(const IterDomain* v) {
+  handle(dynamic_cast<const Val*>(v));
```

Comment:
nit: Is `static_cast` sufficient?

Reply:
Should be. For now, I will just leave it `dynamic_cast` (because I am lazy to change :D), my next PR will remove `IrPrinter` completely.